### ISSUE TITLE
Ref is called twice

### ISSUE THREAD

**Issue by juanluispaz** [developit/preact#98]
The ref function is called twice, the first one with the component instance (as React) and the second one with the DOM node instance

Note: the referred component is an own component (it is no a standard html tag)

**Comment by developit** [comment]
@juanluispaz Thanks for the report, I'll try to reproduce this tonight. The second call is likely being ignored by sinon in the tests.

**Comment by juanluispaz** [comment]
Additional information:

The first call is come from: file: component.js, function: setComponentProps, line: 60, code: hook(props, 'ref', component);

Here 'component' is a component instance

Te second one come from: file: diff.js, function: diff, line: 66, code: (out[ATTR_KEY].ref = originalAttributes.ref)(out);

Here 'out' is a DOM node.

Thanks!

**Comment by developit** [comment]
One more bit to help me debug: is the Component a function, or a Class?

**Comment by juanluispaz** [comment]
Both components are classes

**Comment by juanluispaz** [comment]
Reduced test case:

```javascript
const React = require('react');
const ReactDOM = require('react-dom');

class C extends React.Component {
    render() {
        // Note: en preact this.props is an object with ref property
        // in preact this.props = {ref: ...}
        return <div {...this.props}>hello</div>;
        // But if you replace the last line with:
        // return <div>hello</div>;
        // It will work well
    }
}

class App extends React.Component {
    takeRef = (ref) => {
        console.error('It is called two times with different menu', ref);
    }
    render() {
        return (
            <C ref={this.takeRef} />
        );
    }
}
ReactDOM.render(<App/>, document.getElementById('app'));
```

Note: I am using here react-compact with babel and webpack

**Comment by developit** [comment]
@juanluispaz Ahh, right, so the issue might just be that the ref property is incorrectly being passed down through the Component.

**Comment by developit** [comment]
Fixed and released in [4.5.0](https://github.com/developit/preact/releases/tag/4.5.0).
